keras/backend/mxnet_backend.py

```diff
@@ -2038,8 +2044,12 @@ def concatenate(tensors, axis=-1):
         else:
             axis = 0
 
-    tensors = [t.symbol for t in tensors]
-    return KerasSymbol(mx.sym.concat(*tensors, dim=axis))
+    symbols = [t.symbol for t in tensors]
+
+    if py_all([is_sparse(t) for t in tensors]):
+        return KerasSymbol(mx.sym.sparse.concat(*symbols, dim=0))
```

Comment:
you should still check if axis !=0, even if all tensors are sparse, if user want to concat on axis other than 0, we use dense concat

Reply:
we have hard-coded the dimension - why do we need to check the axis!=0 here?

Reply:
removed hard-coded value & added unit test